src/models/v3/client/profile.rs

```diff
@@ -0,0 +1,146 @@
+#[serde(tag = "game")]
+pub enum ClientProfileMetadata {
+    #[serde(rename = "minecraft-java")]
+    Minecraft {
+        /// Client game version id
+        game_version: String,
+        /// Loader version
+        loader_version: String,
+    },
+    #[serde(rename = "unknown")]
+    Unknown,
+}
+
+impl From<database::models::client_profile_item::ClientProfileMetadata> for ClientProfileMetadata {
+    fn from(game: database::models::client_profile_item::ClientProfileMetadata) -> Self {
+        match game {
+            database::models::client_profile_item::ClientProfileMetadata::Minecraft {
+                loader_version,
+                game_version,
+                ..
+            } => Self::Minecraft {
+                loader_version,
+                game_version,
+            },
+            database::models::client_profile_item::ClientProfileMetadata::Unknown { .. } => {
+                Self::Unknown
+            }
+        }
+    }
+}
+
+impl ClientProfile {
+    pub fn from(
+        profile: database::models::client_profile_item::ClientProfile,
+        current_user_id: Option<database::models::ids::UserId>,
+    ) -> Self {
+        let users = if Some(profile.owner_id) == current_user_id {
+            Some(profile.users.into_iter().map(|v| v.into()).collect())
+        } else {
+            None
+        };
+
+        Self {
+            id: profile.id.into(),
+            owner_id: profile.owner_id.into(),
+            name: profile.name,
+            created: profile.created,
+            updated: profile.updated,
+            icon_url: profile.icon_url,
+            users,
+            loader: profile.loader,
+            game: profile.metadata.into(),
+            versions: profile.versions.into_iter().map(Into::into).collect(),
+            override_install_paths: profile.overrides.into_iter().map(|(_, v)| v).collect(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ClientProfileShareLink {
+    pub id: ClientProfileLinkId, // The url identifier, encoded as base62
+    pub url: String,             // Includes the url identifier, intentionally redundant
+    pub profile_id: ClientProfileId,
+    pub created: DateTime<Utc>,
+    pub expires: DateTime<Utc>,
+}
+
+impl From<database::models::client_profile_item::ClientProfileLink> for ClientProfileShareLink {
+    fn from(link: database::models::client_profile_item::ClientProfileLink) -> Self {
+        // Generate URL for easy access
+        let profile_id: ClientProfileId = link.shared_profile_id.into();
+        let link_id: ClientProfileLinkId = link.id.into();
+        let url = format!(
+            "{}/v3/client/profile/{}/accept/{}",
+            dotenvy::var("SELF_ADDR").unwrap(),
+            profile_id,
+            link_id
+        );
+
+        Self {
+            id: link_id,
+            url,
+            profile_id,
+            created: link.created,
+            expires: link.expires,
+        }
+    }
+}
```

Comment:
The URL shouldn't be returned / parsed here. The frontend will generate it based on the link ID